Limit width of elements

tripu opened the pull request:
&hellip;as suggested in #11.

I've limited the width of all `textarea`s; now they don't push the width beyond the limits of their containers.

One thing is missing, though: I've wrapped all 4 tables here in `<div class="overlarge">`s, and I've been playing with the CSS properties of both those `div`s and the `table`s they contain, to try and limit their width, and instead have horizontal scrollbars appear when they overflow the parent's width. But I didn't manage to achieve that effect&hellip;

@dontcallmedom or @guibbs, care to take a look and tell me [what properties I'm missing](https://github.com/w3c/wbs-design/blob/tripu/fix-width/css/site-wbs-style.css#L39-L45)? The problem is apparent on small screens.

tripu committed:
Limit width of textareas for mobile

Also, remove trailing spaces.

tripu committed:
Wrap tables in a div.overlarge to try limit widths

dontcallmedom commented:
two things:
- I think you want to limit the width of `div.overlarge` to trigger the `overflow-x`; I think `width: 100vw;` should do the trick for that
- more importantly, to make the design really mobile friendly, I would suggest moving away from tables altogether; that requires changes to the markup, but I think it's going to be a lot more effective.

gbaudusseau commented:
There is also some tricks using ::before and/or ::after to make tables responsive. Remember what we tried on systeam pages @tripu? Don't know if @dontcallmedom will like a such hack though :)

tripu committed:
Limit width of ".overlarge" to 100vw

tripu commented:
> I think you want to limit the width of `div.overlarge` to trigger the `overflow-x`; I think `width: 100vw;` should do the trick for that

Yeah, I tried that. [I just committed that change again](https://github.com/w3c/wbs-design/commit/c7975424ae6f7d4dbaa56113348842412cfdb6ca). The problem is, that `div` doesn't take the width of its parent, but more. It looks like margins or paddings somewhere are [pushing the total width over the edge on small screens](https://w3c.github.io/wbs-design/html/qtypes-site.html). I haven't found yet where is that extra margin.

> more importantly, to make the design really mobile friendly, I would suggest moving away from tables altogether; that requires changes to the markup, but I think it's going to be a lot more effective.
> 
> There is also some tricks using ::before and/or ::after to make tables responsive. Remember what we tried on systeam pages @tripu? Don't know if @dontcallmedom will like a such hack though :)

+1 to remove tables for layout.
I'd prefer to get this little fix in place first, and go step by step.

tripu committed:
Don't break the line inside viewport meta tag

tripu commented:
@lcarcone, I have fixed a small detail about the viewport meta tag in one of the HTML files.

I have played a bit with properties of `.overlarge` and tables, but I don't know how to fit tables nicely in narrow screens without more drastic changes to the layout (eg, using `flex` instead of tables; or simply limiting the max width of tables in questionnaires).

I say let's merge this one, and tackle larger changes in a separate PR.

tripu commented:
@lcarcone, OK to merge, then?

lcarcone commented:
@tripu

Hi,

Is there a way to limit the width of the textareas to a % of the container ? ( the 'Comments' and 'Rationale' boxes are taking all the width of the container, I guess they shouldn't be widder than the question choice box).

Also, for questions 11,12, 15 and 16, the textareas and some elements of the question go beyond the limits of their containers. Is it a problem with the markup ?

Thanks,
Laurent

tripu committed:
Fix a paragraph that was commented out incorrectly

tripu committed:
Fix spill of wide tables; adjust borders and backgrounds too

tripu commented:
@lcarcone, I adjusted those widths; check them out.

While at it, I also &ldquo;fixed&rdquo; a few margins and paddings, and also borders and background colours. Items inside questions are now better aligned vertically, there is less wasted space, and the look is simpler now, with fewer borders (which weren't consistent, anyway). The &ldquo;pijama&rdquo; that tables are wearing looks better and makes the text a bit more readable, too.

Let me know what you think!
